packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -80,54 +80,60 @@ class IMobileDevice {
 }
 
 class Xcode {
-  Xcode() {
-    _eulaSigned = false;
+  bool get isInstalledAndMeetsVersionCheck => isInstalled && xcodeVersionSatisfactory;
 
-    try {
-      _xcodeSelectPath = runSync(<String>['xcode-select', '--print-path'])?.trim();
-      if (_xcodeSelectPath == null || _xcodeSelectPath.isEmpty) {
-        _isInstalled = false;
-        return;
-      }
-      _isInstalled = true;
-
-      _xcodeVersionText = runSync(<String>['xcodebuild', '-version']).replaceAll('\n', ', ');
-
-      if (!xcodeVersionRegex.hasMatch(_xcodeVersionText)) {
-        _isInstalled = false;
-      } else {
-        try {
-          final ProcessResult result = processManager.runSync(<String>['/usr/bin/xcrun', 'clang']);
-
-          if (result.stdout != null && result.stdout.contains('license'))
-            _eulaSigned = false;
-          else if (result.stderr != null && result.stderr.contains('license'))
-            _eulaSigned = false;
-          else
-            _eulaSigned = true;
-        } catch (error) {
-          _eulaSigned = false;
-        }
+  String _xcodeSelectPath;
+  String get xcodeSelectPath {
+    if (_xcodeSelectPath == null) {
+      try {
+        _xcodeSelectPath = runSync(<String>['/usr/bin/xcode-select', '--print-path'])?.trim();
+      } on ProcessException {
+        // Ignore: return null below.
       }
-    } catch (error) {
-      _isInstalled = false;
     }
+    return _xcodeSelectPath;
   }
 
-  bool get isInstalledAndMeetsVersionCheck => isInstalled && xcodeVersionSatisfactory;
-
-  String _xcodeSelectPath;
-  String get xcodeSelectPath => _xcodeSelectPath;
-
-  bool _isInstalled;
-  bool get isInstalled => _isInstalled;
+  bool get isInstalled {
+    if (xcodeSelectPath != null && xcodeSelectPath.isNotEmpty)
+      return false;
```

Comment:
Whoops - tried to get clever, realised it wouldn't work and missed a revert.  Once this is in, asyncified and the process invocations extracted, we'll be able to easily add tests for these cases.